src/mcpd/mcpd_client.py

```diff
@@ -423,7 +433,16 @@ def agent_tools(self) -> list[Callable[..., Any]]:
         agent_tools = []
         all_tools = self.tools()
 
-        for server_name, tool_schemas in all_tools.items():
+        # Determine which servers to use.
+        servers_to_use = all_tools.keys() if servers is None else servers
+
+        # Fetch tools from selected servers.
+        for server_name in servers_to_use:
+            if server_name not in all_tools:
+                # Server doesn't exist or has no tools - skip silently.
```

Comment:
Not sure about the skip silently here, should probably raise a warning

Reply:
Yeah, I know where you're coming from. It was initially just to have parity with the JS SDK (https://github.com/mozilla-ai/mcpd-sdk-javascript/blob/main/src/client.ts#L700-L702) but I will take another look at them both around this.